lib/jormungandr/src/Cardano/Wallet/Jormungandr/Binary.hs

```diff
@@ -744,11 +823,14 @@ putAddress (Address bs) = putByteString bs
 --                  / %x07 POOL-UPDATE
 --                  / %x08 UPDATE-PROPOSAL
 --                  / %x09 UPDATE-VOTE
-withHeader :: FragmentSpec -> Put -> Put
-withHeader spec content = do
-    let bs = BL.toStrict $ runPut (putFragmentSpec spec *> content)
-    putWord16be (toEnum $ BS.length bs)
-    putByteString bs
+withHeader :: FragmentSpec -> Put -> (Hash "Tx", SealedTx)
+withHeader spec content =
```

Comment:
This completely changes the semantic of the original function which was making no particular assumption of the fragment type and was doing one and one thing. Now, the function does enforce a very specific output and does also computes an extra hash?

Reply:
The `Hash "Tx"` is *the* `fragmentId` which makes no assumption about the fragment type.

The main difference is returning a `SealedTx` instead of a `Put`. We have no need for returning a `Put`, and this seemed like an appropriate function to construct the actual `SealedTx`.

We could rename it. We could also return `(Hash "Fragment", ByteString)` or something instead.